test/integration/master/kms_transformation_test.go

```diff
@@ -140,11 +123,79 @@ func TestKMSProvider(t *testing.T) {
 	}
 
 	// Secrets should be un-enveloped on direct reads from Kube API Server.
-	s, err := test.restClient.CoreV1().Secrets(testNamespace).Get(testSecret, metav1.GetOptions{})
+	mustReadSecret(t, test, true)
+	test.printMetrics()
+}
+
+// TestEnvelopCache validates that KMS Plugin will not be called for secrets that were cached by api-server\pkg\storage\cacher.go
+// Cacher.go (assuming sufficient memory) should intercept requests for secrets that were recently created or recently retrieved - thus
+// alleviating the need for caching within envelop.go.
+func TestEnvelopCache(t *testing.T) {
+	pluginMock, test := mustSetupTest(t)
+	defer pluginMock.cleanUp()
+	defer test.cleanUp()
+
+	// Expect hit - cache was populated when the test secret was created (i.e. cacher.go has a cached version of the secret in plain text).
+	mustReadSecret(t, test, false)
+
+	select {
+	case d := <-pluginMock.decryptRequest:
+		t.Fatalf("got %v, expected nil, read request did not come from cache.", d)
+	default:
+		t.Logf("read of %s, was handled via cache", test.secret.Name)
```

Comment:
no need for this log, just leave a blank default case

Reply:
Done.